cmd/kubeadm/app/preflight/checks.go

```diff
@@ -148,10 +148,16 @@ func (sc ServiceCheck) Check() (warnings, errors []error) {
 				sc.Service, sc.Service))
 	}
 
-	if sc.CheckIfActive && !initSystem.ServiceIsActive(sc.Service) {
+	if sc.WantActive != initSystem.ServiceIsActive(sc.Service) {
+		status := [2]string{"active", "not active"}
+		command := [2]string{"stop", "start"}
```

Comment:
This parsing of intent from the return is odd to me.  I'd think that having a utility function, or augmenting the return arg would be cleaner.

Reply:
I've changed this.